integration/spark/src/main/scala/org/apache/carbondata/indexserver/DistributedRDDUtils.scala

```diff
@@ -164,6 +177,14 @@ object DistributedRDDUtils {
     }
   }
 
+  def isSegmentInProgress(request: IndexInputFormat, segment: String): Boolean = {
+    request.getReadCommittedScope.getSegmentList.find(_.getLoadName
+      .equalsIgnoreCase(segment)) match {
+      case Some(value) => value.getSegmentStatus.equals(SegmentStatus.INSERT_IN_PROGRESS)
```

Comment:
may be need to handle INSERT_OVERWRITE_IN_PROGRESS segment also here

Reply:
done